### PR TITLE
docs(contributing): add CONTRIBUTING.md with guidelines for contributors

### PR DESCRIPTION
**Pull request by dylanpulver**
## Description

This pull request adds a `CONTRIBUTING.md` file to the repository, outlining the guidelines for contributing to the SafetyCLI project. The document provides details on how to report bugs, suggest enhancements, improve documentation, and submit code changes. It also includes information on the development setup, running tests, and the pull request process.

### Key Changes:
- Added a `CONTRIBUTING.md` file with sections on how to contribute, development setup, and pull request process.
- Defined labels for issues such as `bug`, `enhancement`, `documentation`, and more.
- Provided a link to the project’s Code of Conduct and License.
- Updated the documentation link to point to the correct location: https://docs.safetycli.com/safety-docs.
- Included instructions for running tests and guidelines for commit messages based on Conventional Commits.

## Checklist:
- [x] The contributing guidelines have been thoroughly reviewed.
- [x] Links and labels are correctly set and functional.
- [x] The document follows the project's style and tone.

## Related Issues
N/A

## Additional Notes
This document aims to provide a clear and structured way for contributors to participate in the project. Any feedback or further suggestions to improve the guidelines are welcome.

## Summary by CodeRabbit

- **New Features**
	- Introduced a comprehensive `CONTRIBUTING.md` file to guide users on how to contribute to the SafetyCLI project.
	- Included sections on bug reporting, enhancement suggestions, documentation improvements, and development setup.
	- Established guidelines for adherence to a code of conduct and commit message formatting.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  - [Pull Request Process](#pull-request-process)
+    - [Commit Messages](#commit-messages)
+    - [Merging Changes](#merging-changes)
+  - [License](#license)
+
+## How Can I Contribute?
+
+### Reporting Bugs
+
+If you find a bug, please report it by opening a [GitHub issue](https://github.com/pyupio/safety/issues). When reporting a bug, please include:
```
Comment (davidlacho) [inline CONTRIBUTING.md:25]: Is it worth linking to our bug bounty page?

**Review by davidlacho — APPROVED** [review]
💅🏻

File: CONTRIBUTING.md (hunk 1 above)
Comment (dylanpulver) [reply CONTRIBUTING.md:25]: For sure! I just added that in!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+- `help wanted`: Requests assistance from the community or other contributors.
+- `good first issue`: Highlights issues that are suitable for first-time contributors.
+
+## Getting Started
+
+### Code of Conduct
+
+Please read and adhere to our [Code of Conduct](CODE_OF_CONDUCT.md) to ensure a welcoming environment for all contributors.
+
+###  Development Setup
```
[the remark this answers is not in the document]
Comment (dylanpulver) [reply CONTRIBUTING.md]: sure!

**Pull request merged**